core/eolearn/core/eoworkflow.py

```diff
@@ -308,6 +313,15 @@ def get_nodes(self) -> List[EONode]:
         """
         return self._nodes[:]
 
+    def get_node_with_uid(self, uid: Optional[str], fail_if_missing: bool = False) -> Optional[EONode]:
+        """Returns node with give uid, if it exists in the workflow."""
+        relevant_nodes = [node for node in self._nodes if node.uid == uid]
```

Comment:
Why not just checking if `uid` exists in `self._uid_dict`? This way you achieve O(1) time complexity instead of O(n) and uniqueness is ensured.

Reply:
oh yes, ofcourse, I now have the `_uid_dict` at disposal. Will change